modules/desktop/graphics/ewwbar.nix

```diff
@@ -276,29 +278,39 @@ let
           fi
       }
 
-      set_screen() {
-          brightnessctl set "$1""%" -q
-      }
-
       get() {
-          brightness=$(screen_level)
+          brightness=$(brightnessctl info | grep -oP '(?<=\().+?(?=%)' | awk '{print $1 + 0.0}')
           icon=$(icon "$brightness")
           echo "{ \"screen\": { \"level\": \"$brightness\" }, \"icon\": \"$icon\" }"
       }
 
       listen() {
-        inotifywait -m /sys/class/backlight/*/brightness | \
-        while read -r line; do
-            if echo "$line" | grep -q "CLOSE_WRITE"; then
+        prev_brightness=""
+        inotifywait -m -e close_write /sys/class/backlight/*/brightness |
+        while read -r; do
+            get &
+            current_brightness=$(brightnessctl info | grep -oP '(?<=\().+?(?=%)' | awk '{print $1 + 0.0}')
```

Comment:
How many times this loop runs per sec?
It calls external app(s) in a loop, hopefully for not too long time?

Reply:
It's quite responsive in use and hasn't caused issues yet.
Volume control (and listening to volume changes) is still less responsive than brightness overall.

Reply:
We can of course remove the checking if value has changed, but that will cause the brightness popup to appear even if, for example, the brightness is being increased while it's already at 100%

Reply:
I'm not an expert in this, i just saw a loop to do some fairly heavy calls for updating which may cause lots of load.
Yes in most cases this is likely to be finished in one loop, not sure if it is possible to make it run for longer period of time and cause lots of resource usage.
In the loop there is get in background process and running same brightnessctl again in the next line.
Would it be possible to combine getting the brightnessctl content in one call and reuse the data?

Reply:
In practice, the inside of the while loop gets executed only when a **close_write** event is detected in **/sys/class/backlight/*/brightness**.
This can be checked manually by running **`/nix/store/k5icwhkqvk77zqksi6kld2mjpywql7qg-eww-brightness/bin/eww-brightness listen`** and adjusting display brightness using F5 and F6.

Reusing the data from **brightnessctl** in the **get** call is possible, we could provide it as an argument, but then we'd end up needing an if/else check inside of **get()**.
We would also need to fetch the value before calling **get**, which would cause a minor delay and the overall responsiveness might be affected.

That being said, I'll look into it along with the other changes to be done.

Reply:
Implemented.
The result is slightly (few ms) better responsiveness.